helix-view/src/editor/variable_expansion.rs

```diff
@@ -0,0 +1,159 @@
+
+                                            let output = tokio::task::block_in_place(move || {
+                                                helix_lsp::block_on(async move {
+                                                    let mut command =
+                                                        tokio::process::Command::new(&shell[0]);
+                                                    command.args(&shell[1..]).arg(&body[..]);
+
+                                                    let output =
+                                                        command.output().await.map_err(|_| {
+                                                            anyhow::anyhow!(
+                                                                "Shell command failed: {body}"
+                                                            )
+                                                        })?;
+
+                                                    if output.status.success() {
+                                                        String::from_utf8(output.stdout).map_err(
+                                                            |_| {
+                                                                anyhow::anyhow!(
+                                                            "Process did not output valid UTF-8"
+                                                        )
+                                                            },
+                                                        )
+                                                    } else if output.stderr.is_empty() {
+                                                        Err(anyhow::anyhow!(
+                                                            "Shell command failed: {body}"
+                                                        ))
+                                                    } else {
+                                                        let stderr =
+                                                            String::from_utf8_lossy(&output.stderr);
+
+                                                        Err(anyhow::anyhow!("{stderr}"))
+                                                    }
+                                                })
+                                            });
+                                            o.push_str(&input[last_push_end..index]);
+                                            last_push_end = end + 1;
+
+                                            o.push_str(output?.trim());
+
+                                            break;
+                                        }
+                                    }
+                                } else if char == '{' {
+                                    right_bracket_remaining += 1;
+                                }
+                            }
+                        }
+                    }
```

Comment:
btw I didn't know why we are not using regex here. I think it could make the code way much clearer, but I kept it the way it was done in the original PR (my comment is about the whole file)